packages/frontend/editor-ui/src/views/SigninView.test.ts

```diff
@@ -97,4 +106,76 @@ describe('SigninView', () => {
 
 		expect(router.push).toHaveBeenCalledWith('/home/workflows');
 	});
+
+	describe('when redirect query parameter is set', () => {
+		const ORIGIN_URL = 'https://n8n.local';
+		let route: ReturnType<typeof useRoute>;
+
+		beforeEach(() => {
+			route = useRoute();
+			global.window = Object.create(window);
+
+			Object.defineProperty(window, 'location', {
+				value: {
+					href: '',
+					origin: ORIGIN_URL,
+				},
+				writable: true,
+			});
```

Comment:
Allow mocking of `window.location.origin` and allow us to spy on `window.location.href`.